### ISSUE TITLE
url problem (non standard) any advice?

### ISSUE THREAD

**Issue by rhysburnie** [jtangelder/sass-loader#260]
Hello,

I have an issue in regards to loading background images, it's not the same issues 
mentioned under **Problems with url(...)** in the README.

To understand the issue I need to briefly explain the setup:

The platform (of distributed touchscreen kiosks) I am developing for requires
that the final production build is delivered over `file://` protocol, i.e. it is
not run on a local server or any server it's exactly like opening a html file 
via Finder or Explorer.

As you can imagine this is quite frustrating but I have managed to setup webpack
so the build does work.

In this have an `assets` folder with `css`, `js`, `images`. The webpack server serves 
`assets` on path `/static/`. In production the build loads assets via the path 
`./static/`.

To be clear all js, css and image src load correctly in both environments.

The issue is due to the way css `background-image: url(...)` is written.

I use `url(~assets/images/image.jpg)` in the css, this works great in development.

In a normal webpack project (one that would expect its served on a real server) 
you would expect the css output to be: `url(/static/img/image.jpg)`.

Now here's the problem... because I have to change the value of `/static/` in 
production to a non standard `./static/` the sass-loader does its thing correctly
i.e. `url(./static/img/image.jpg)`, However because it needs to be served via 
local paths it needs to be `url(../img/image.jpg)`.

I realize that this is not a bug of the loader as it's technically working correctly,
but I'm a bit stuck about how to go about getting what I need. I suspect that I 
may need to add some kind of additional custom process after the sass loader that
replaces string instances of `url(./static/` to `url(../` but I don't know how.

**Does anyone have any advice on how I can go about replacing this path in production?**

Like I said all paths work for the entire app, js, css, src of image elements the
only issue it the paths in background-image.

Any help would be much appreciated

Cheers

**Comment by jhnns** [comment]
This sounds very complicated and I don't think that it should be that way. My advice:

- Don't serve your HTML over `file://`. There are a lot of cases where things behave differently and you will run in more issues than this. Do yourself a favor and serve your stuff over an HTTP server.
- If you can't serve your stuff over HTTP by all means, you should **not** use the webpack-dev-server for development. Your development environment should be as close as possible to the production environment and `file://` vs `http://` is a **huge** difference.
- If you **really really** need to use the `file://` protocol: Why are you using webpack? Just split your stuff and put it somewhere on the disk. Webpack is a bundler and bundler put a lot of stuff into a few files to serve it efficiently over **HTTP**. If you don't have a roundtrip, I doubt that you need webpack at all.
- My last advice: Maybe the [`<base>`](https://developer.mozilla.org/en/docs/Web/HTML/Element/base) tag can help you.

**Comment by rhysburnie** [comment]
I knew the reply would be "dont use `file://`"...

Why would I choose to use `file://`? this isn't the first time I've developed an app. The system I have to use for this network of distributed devices forces me to do so, I think its stupid also.

I am using webpack because the app is huge and there is an entire ecosystem of tools that work great for building apps which work much better and easier via a build systems like webpack or gulp.

It's only this one little issue that didn't work and I have now resolved it.

I tried various things but each solution would tend to mess with other parts of the app so finally I just ended up adding an additional script to run after build which uses the same app config settings to do a sed replacement on the css file.
